packages/devtools_app/lib/src/shared/banner_messages.dart

```diff
@@ -479,6 +479,67 @@ For the most accurate absolute memory stats, relaunch your application in ''',
   }
 }
 
+class DebuggerIdeRecommendationMessage {
+  const DebuggerIdeRecommendationMessage(this.screenId);
+
+  final String screenId;
+
+  BannerMessage build(BuildContext context) {
+    final codeType =
+        (serviceConnection.serviceManager.connectedApp?.isFlutterAppNow ??
+                false)
+            ? 'Flutter'
+            : 'Dart';
+    final g3FlutterInstructionsLink =
+        devToolsExtensionPoints.g3FlutterIdeRecommendationLink();
+    final g3DartInstructionsLink =
+        devToolsExtensionPoints.g3FlutterIdeRecommendationLink();
+
+    return BannerWarning(
+      key: Key('DebuggerIdeRecommendationMessage - $screenId'),
+      textSpans: [
+        TextSpan(
+          text: '''
+The $codeType DevTools debugger is in maintenance mode. For a best-in-class debugging experience, we recommend debugging your $codeType code in an IDE, such as ''',
+        ),
+        if (g3FlutterInstructionsLink != null) ...[
+          GaLinkTextSpan(
+            context: context,
+            link: g3FlutterInstructionsLink,
+          ),
+        ] else if (g3DartInstructionsLink != null) ...[
+          GaLinkTextSpan(
+            context: context,
+            link: g3DartInstructionsLink,
+          ),
+        ] else ...[
+          GaLinkTextSpan(
+            context: context,
+            link: const GaLink(
+              display: 'VS Code',
+              url: 'https://dart.dev/tools/vs-code',
+            ),
+          ),
+          const TextSpan(
+            text: ' or ',
+          ),
+          GaLinkTextSpan(
+            context: context,
+            link: const GaLink(
+              display: 'IntelliJ & Android Studio',
+              url: 'https://dart.dev/tools/jetbrains-plugin',
+            ),
+          ),
+        ],
+        const TextSpan(
+          text: '.',
```

Comment:
nit: drop the trailing comma on these textspans that will fit on one line

Reply:
Done!